### PR TITLE
fix(Partner Center Sell): re-gen service after recent API definition changes

### PR DESCRIPTION
**Pull request by HarasztiaPeter**
## PR Checklist
Please make sure that your PR fulfills the following requirements:  
- [x] The commit message follows the [Angular Commit Message Guidelines](https://github.com/angular/angular/blob/master/CONTRIBUTING.md#-commit-message-guidelines).
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## Does this PR introduce a breaking change?    
- [ ] Yes
- [x] No

### CODE REVIEW

**Comment by HarasztiaPeter** [comment]
The docs were updated as well under the https://test.cloud.ibm.com/apidocs/partner-center-sell?code=go link

**Comment by HarasztiaPeter** [comment]
Screenshot 2024-10-02 at 14 25 43
Test run

**Comment by HarasztiaPeter** [comment]
Link to yaml https://product-lifecycle.api.test.cloud.ibm.com/openapi/v1/openapi.yaml

**Review by padamstx — APPROVED** [review]

**Review by padamstx — COMMENTED** [review]
Removed my approval until some of the comments have been reviewed and a decision is made as to whether or not to address them.

File: partnercentersellv1/partner_center_sell_v1.go [hunk 1]
```diff
@@ -3155,13 +3045,13 @@ type CatalogHighlightItem struct {
 	Description *string `json:"description,omitempty"`
 
 	// The description about the features of the product in translation.
-	DescriptionI18n map[string]string `json:"description_i18n,omitempty"`
+	DescriptionI18n map[string]interface{} `json:"description_i18n,omitempty"`
```
[the remark this answers is not in the document]
Comment (HarasztiaPeter) [reply partnercentersellv1/partner_center_sell_v1.go:3048]: We had issue with passing custom data, in terraform this change seems to solve that problem.

File: partnercentersellv1/partner_center_sell_v1_integration_test.go [hunk 2]
```diff
@@ -349,43 +279,52 @@ var _ = Describe(`PartnerCenterSellV1 Integration Tests`, func() {
 
 			catalogHighlightItemModel := &partnercentersellv1.CatalogHighlightItem{
 				Description:     core.StringPtr("highlight desc"),
-				DescriptionI18n: map[string]string{"key1": "desc"},
+				DescriptionI18n: map[string]interface{}{"key1": "testString"},
```
[the remark this answers is not in the document]
Comment (HarasztiaPeter) [reply partnercentersellv1/partner_center_sell_v1_integration_test.go:282]: Its true, but right now we do not, plan to go into other languages, the main use case for thisservice is still terraform. Somewhere in the docs we should add, it has been added to related terraform docs as well.
**Note - Intended for internal use only. This resource is strictly experimental and subject to change without notice.**

File: partnercentersellv1/partner_center_sell_v1_integration_test.go [hunk 3]
```diff
@@ -401,8 +340,8 @@ var _ = Describe(`PartnerCenterSellV1 Integration Tests`, func() {
 
 			supportDetailsItemAvailabilityTimeModel := &partnercentersellv1.SupportDetailsItemAvailabilityTime{
 				Day:       core.Float64Ptr(float64(72)),
-				StartTime: core.StringPtr("2024-01-01T12:00:00.000Z"),
```
[the remark this answers is not in the document]
Comment (HarasztiaPeter) [reply partnercentersellv1/partner_center_sell_v1_integration_test.go:404]: Yes it was intentional, it should accept time like 10:30, not just date time

File: partnercentersellv1/partner_center_sell_v1_integration_test.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (padamstx) [reply partnercentersellv1/partner_center_sell_v1_integration_test.go:282]: You could perhaps update the service table in README.md to include `Experimental` after the service name:
image

You could perhaps also add `Note: the Partner Center Sell service is intended for internal use only. This service is strictly experimental and subject to change without notice.` just after the table of "moved" services:
image
(right before Prerequisites)

File: partnercentersellv1/partner_center_sell_v1.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (padamstx) [reply partnercentersellv1/partner_center_sell_v1.go:3048]: What do you mean specifically by "custom data"?  Does the service support values of any type (string, boolean, integer, object, float, etc.)?

File: partnercentersellv1/partner_center_sell_v1_integration_test.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (padamstx) [reply partnercentersellv1/partner_center_sell_v1_integration_test.go:282]: > Its true, but right now we do not plan to go into other languages.

Even if you are only going to produce a Go SDK and Terraform provider, the ability to generate the integration tests and examples code when a change is made to the API (without needing to make manual changes) will save you lots of time in the future.

File: partnercentersellv1/partner_center_sell_v1_integration_test.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (padamstx) [reply partnercentersellv1/partner_center_sell_v1_integration_test.go:404]: Here's a link to the API Handbook where it talks about compliant data types:
https://cloud.ibm.com/docs/api-handbook?topic=api-handbook-types

Has this service's API definition completed a UX029 review yet?

File: partnercentersellv1/partner_center_sell_v1_examples_test.go [hunk 4]
```diff
@@ -40,7 +38,13 @@ import (
 // PARTNER_CENTER_SELL_AUTH_TYPE=iam
 // PARTNER_CENTER_SELL_APIKEY=<IAM apikey>
 // PARTNER_CENTER_SELL_AUTH_URL=<IAM token service base URL - omit this if using the production environment>
-//
+// PRODUCT_ID_APPROVED=<product id>
```
Comment (padamstx) [inline partnercentersellv1/partner_center_sell_v1_examples_test.go:41]:
```suggestion
// PARTNER_CENTER_SELL_PRODUCT_ID_APPROVED=<product id>
```

**Review by padamstx — APPROVED** [review]
I posed a few new questions, but I'm approving now

**Review by padamstx — CHANGES_REQUESTED** [review]
A couple of minor changes needed.  Re: the other questions, I don't think they're showstoppers, so just let me know if you want to ignore for now and I can merge.

File: partnercentersellv1/partner_center_sell_v1_integration_test.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (HarasztiaPeter) [reply partnercentersellv1/partner_center_sell_v1_integration_test.go:404]: UX029 has been completed, thats where we were approved for Experimental.
https://github.ibm.com/CloudEngineering/api-standards/issues/303

File: partnercentersellv1/partner_center_sell_v1_integration_test.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (HarasztiaPeter) [reply partnercentersellv1/partner_center_sell_v1_integration_test.go:282]: If you don't mind the example fix would go to the next PR.

File: partnercentersellv1/partner_center_sell_v1.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (HarasztiaPeter) [reply partnercentersellv1/partner_center_sell_v1.go]: Readded types

File: partnercentersellv1/partner_center_sell_v1.go [hunk 5]
```diff
@@ -42,7 +42,7 @@ type PartnerCenterSellV1 struct {
 }
 
 // DefaultServiceURL is the default URL to make service requests to.
-const DefaultServiceURL = "https://product-lifecycle.api.test.cloud.ibm.com/openapi/v1"
+const DefaultServiceURL = "http://localhost:5005/openapi/v1"
```
Comment (padamstx) [inline partnercentersellv1/partner_center_sell_v1.go:45]: The default service URL should come directly from the "servers" field in the API definition.  It should reflect the production endpoint for the service.

File: catalogmanagementv1/catalog_management_v1_integration_test.go [hunk 6]
```diff
@@ -737,9 +737,9 @@ var _ = Describe(`CatalogManagementV1 Integration Tests`, func() {
 			}
 
 			supportTimeModel := &catalogmanagementv1.SupportTime{
-				Day:       core.Int64Ptr(int64(38)),
-				StartTime: core.StringPtr("testString"),
-				EndTime:   core.StringPtr("testString"),
+				Day:       core.Int64Ptr(int64(30)),
```
Comment (padamstx) [inline catalogmanagementv1/catalog_management_v1_integration_test.go:740]: Just curious how this change is related to the Partner Center Sell service???

**Review by padamstx — COMMENTED** [review]
Getting closer.  Just need to resolve the default service URL and I was just curious about the change to the CM int test.

File: catalogmanagementv1/catalog_management_v1_integration_test.go (hunk 6 above)
Comment (HarasztiaPeter) [reply catalogmanagementv1/catalog_management_v1_integration_test.go:740]: Sorry i do not understand the question can you ellaborate?

**Pull request merged**